src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -381,8 +395,24 @@ private static unsafe bool TryParseNumber(ref char* str, char* strEnd, NumberSty
             }
 
             bool negExp = false;
-            number.DigitsCount = digEnd;
+            if (digEnd == maxDigCount && numberOfTrailingZeros > 0 && number.HasNonZeroTail)
+            {
+                // We have an additional non-zero digit in the input past maxDigitCount. We already handle this properly.
+                number.DigitsCount = digEnd;
```

Comment:
So in this situation, we are maintaining the trailing zeros in the buffer? Won't that conflict with the consistency check?

Reply:
It should've. It probably didn't because of the way I was comparing `byte` and `0`. Fixed now.